### PR TITLE
fanout fixture in test_drop_counters replaced with FanoutHost

### PR DESCRIPTION
**Pull request by andriyz-nv**
### Description of PR
Changed fanout fixture to relay on type of FanoutHost. This allows to 
provide vendor specific implementation of sending broken packets.
This fixture is used in test_equal_smac_dmac_drop, test_multicast_smac_drop
This tests are skipped on mellanox hwsku with SONIC OS because drop counters
are not shown in SONIC for this reason 

Summary:
Fixes # https://github.com/Azure/sonic-mgmt/issues/3383

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [x] 201911

#### What is the motivation for this PR?
This change let framework decide which configuration need to be applied for fanout switch

#### How did you verify/test it?
SONiC Software Version: SONiC.202012.141-e62ae02fb_Internal
Distribution: Debian 10.10
Kernel: 4.19.0-12-2-amd64
Build commit: e62ae02fb
Build date: Tue Aug 10 18:06:03 UTC 2021
Built by: sw-r2d2-bot@r-build-sonic-ci02-242

Platform: x86_64-mlnx_msn4700-r0
HwSKU: ACS-MSN4700
ASIC: mellanox
ASIC Count: 1
Serial Number: MT2022X08595
Uptime: 14:35:03 up  9:39,  1 user,  load average: 0.19, 0.19, 0.30

### CODE REVIEW

**Comment by AndriyZabavskyy** [comment]
@wangxin  can you review, please?

**Comment by liat-grozovik** [comment]
@yxieca could you please assign someone to review?

File: tests/common/devices/onyx.py [hunk 1]
```diff
@@ -196,13 +208,85 @@ def get_speed(self, interface_name):
...
+            openflow_port_id=openflow_port_id, set_mac=set_mac, eth_field=eth_field)])
+        if 'failed' in out and out['failed']:
+            logger.error('Failed to set openflow rule - {}'.format(out['msg']))
+            return False
+        logger.debug('Setting openflow rule succed from onyx: {}'.format(out))
+        return True
+
+    def get_fanout_facts(self):
+        result = self.localhost.conn_graph_facts(host=self.hostname,
+                                                 filepath=LAB_CONNECTION_GRAPH_DIR)
```
Comment (yxieca) [inline tests/common/devices/onyx.py:250]: filepath is not the preferred way to choose right graph file. in conn_graph_facts.py, there is already automation to choose a graph file to best suite the device(s) passed in. Is that method failing for you?

Comment (yxieca) [reply tests/common/devices/onyx.py:250]: Do you know why? lab file is the first one checked in the list: https://github.com/Azure/sonic-mgmt/blob/master/ansible/files/graph_files.yml#L3

Can you make the existing mechanism working.

By merging this PR, nightly test depends on the existing automation will fail.

Comment (andriyz-nv) [reply tests/common/devices/onyx.py]: @yxieca I did changed the implementation of get_fanout_facts() to fixture fanout_graph_facts

**Comment by andriyz-nv** [comment]
Hi,
Can anyone help to understand what is wrong with container_checker in Pre-test sanity?

`>                       .format(json.dumps(failed_results, indent=4, default=fallback_serializer)))
E                   Failed: !!!!!!!!!!!!!!!!Pre-test sanity check failed: !!!!!!!!!!!!!!!!
E                   [
E                       {
E                           "check_item": "monit", 
E                           "failed": true, 
E                           "host": "vlab-03", 
E                           "services_status": {
E                               "vlab-03": "Running", 
E                               "container_memory_telemetry": "Status ok", 
E                               "diskCheck": "Status ok", 
E                               "root-overlay": "Accessible", 
E                               "container_checker": "Status failed", 
E                               "vnetRouteCheck": "Status ok", 
E                               "var-log": "Accessible", 
E                               "rsyslog": "Running", 
E                               "routeCheck": "Status ok"
E                           }
E                       }
E                   ]
`

**Comment by liat-grozovik** [comment]
@andriyz-nv please handle conflicts.
@yxieca could you please help to review following the changes on the PR?

File: tests/common/helpers/drop_counters/fanout_onyx_drop_counter.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+INVOCATION = 'invocation'
+STDOUT = 'stdout'
+
+
+class FanoutOnyxDropCounter:
+    def __init__(self, onyx_switch):
+        self.onyx_switch = onyx_switch
+        self.fanout_graph_facts = None
+
+    def prepare_config(self, fanout_graph_facts, match_mac, set_mac, eth_field):
```
Comment (lolyu) [inline tests/common/helpers/drop_counters/fanout_onyx_drop_counter.py:21]: Suggest defining a base class to define `prepare_config` and `restore_config` APIs and have those specific implementations in the derivatives.

File: tests/common/devices/onyx.py [hunk 3]
```diff
@@ -196,13 +196,41 @@ def get_speed(self, interface_name):
...
+            fanout_graph_facts (dict): fixture fanout_graph_facts
+            match_mac (str): mac address to match in openflow rule
+            set_mac (str): mac address to which match mac should be changed
+            eth_field (str): place in which replace match mac to set_mac, usually 'eth_src'
+
+        Returns:
+            boolean: True if success. Usually, the method return False only if the operation
+            is not supported or failed.
+        """
+        fanout_helper = FanoutOnyxDropCounter(self)
```
Comment (lolyu) [inline tests/common/devices/onyx.py:224]: can we share the `FanoutOnyxDropCounter` object between `prepare_drop_counter_config` and `restore_drop_counter_config`?

**Comment by roysr-nv** [comment]
This PR is replaced by:
https://github.com/Azure/sonic-mgmt/pull/5291

**Pull request closed without merging**